src/main/java/com/fundynamic/d2tm/game/drawing/DrawableViewPort.java

```diff
@@ -0,0 +1,85 @@
+package com.fundynamic.d2tm.game.drawing;
+
+import com.fundynamic.d2tm.Game;
+import com.fundynamic.d2tm.game.Viewport;
+import com.fundynamic.d2tm.game.map.Map;
+import com.fundynamic.d2tm.game.map.Perimeter;
+import com.fundynamic.d2tm.game.math.Vector2D;
+import com.fundynamic.d2tm.graphics.Tile;
+import org.newdawn.slick.Graphics;
+import org.newdawn.slick.SlickException;
+
+public class DrawableViewPort {
+
+    private final Viewport viewport;
+    private final Graphics graphics;
+
+    private final Vector2D<Integer> drawingVector;
+    private Vector2D<Float> viewingVector;
+
+    private final Perimeter<Float> viewingVectorPerimeter;
+    private float velocityX;
+    private float velocityY;
+    private float moveSpeed;
+
+    public DrawableViewPort(Vector2D screenResolution, Viewport viewport, Vector2D viewingVector, Graphics graphics, float moveSpeed) {
```

Comment:
@arjenvanderende this is what I mean with the weird dependencies. `Viewport` also has a screen height and width